cpp/src/gandiva/regex_functions_holder_test.cc

```diff
@@ -252,29 +224,25 @@ TEST_F(TestLikeHolder, TestEmptyEscapeChar) {
 }
 
 TEST_F(TestLikeHolder, TestMultipleEscapeChar) {
-  std::shared_ptr<LikeHolder> like_holder;
-
-  auto status = LikeHolder::Make("ab\\_", "\\\\", &like_holder);
+  auto status = LikeHolder::Make("ab\\_", "\\\\").status();
   EXPECT_EQ(status.ok(), false) << status.message();
 }
 
 class TestILikeHolder : public ::testing::Test {
  public:
   RE2::Options regex_op;
-  FunctionNode BuildILike(std::string pattern) {
+  static FunctionNode BuildILike(std::string pattern) {
```

Comment:
Why do we need this `static`?
I think that we can call this as an instance method in `TEST_F()` without `static`.

Reply:
I follow clang-tidy's suggestion to make this change because this function uses none of the member variable of class `testILikeHolder`, which means it can simply be a static function but not a member function, and the `TEST_F()` can still use them. But it is probably easier to follow if we move them out of the class entirely. Anyway, I change this part back since it is not critical in this PR.